Source/UI/Validator/PhoneNumberValidator.swift

```diff
@@ -1,12 +1,11 @@
 import Foundation
 
 class PhoneNumberValidator: Validator {
-    func validate(text: String?) -> Bool {
-        isEmpty(text: text)
-    }
+    //  https://api-reference.checkout.com/#operation/requestAToken!path=0/phone&t=request
 
-    private func isEmpty(text: String?) -> Bool {
-        text?.isEmpty ?? false
+    func validate(value: Any?) -> Bool {
+        guard let phoneNumber = (value as? Phone) else { return false }
+        return PhoneValidator().validate(phoneNumber) == .success
```

Comment:
The url reference might make more sense if placed inside PhoneValidator. I assume its meant to justify why the current validation is the way it is, which would make sense to happen at the actual validation site.

Also, if we expect this to be called multiple times, we may as well allocate the PhoneValidator internally to save it from being created on each call.

```swift
class PhoneNumberValidator: Validator {
    private let validator = PhoneValidator()

    func validate(value: Any?) -> Bool {
         guard let phoneNumber = (value as? Phone) else { return false }
         return validator.validate(phoneNumber) == .success
    }
}
```